### PR TITLE
Isotope pattern storage

### PR DESCRIPTION
**Pull request by lomereiter**
Isotope patterns are stored in a directory structure `<instrument settings>/<charge>/<adduct>.parquet` so that loading a subset for a job is trivial.

- [x] endpoint for fetching data (updates are rare and can be done manually)
- [x] `batch_generate` method for adding a database/adduct/instrument (existing adducts/instrument settings are 'stored' in filenames)
- [x] `batch_generate` uses multiprocessing for quick computation (one file = one task)
- [x] synchronization with S3 (via AWS CLI tool)
- [x] endpoint for fetching subsampled table (for an sm-engine job), encoding target-decoy pairs in boolean columns, one per target adduct; for target adduct rows the booleans are all zero
- [x] config file entry: path to the storage directory

### CODE REVIEW

File: app/isotope_storage.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        df = self._load(dir_path)
+        if adducts:
+            adducts_df = pd.DataFrame({'adduct': list(set(adducts))})
+            df = pd.merge(df, adducts_df)  # faster than .isin() method
+        if not db_id:  # = all databases
+            return df
+        mol_formulas = self._molecular_formulas(db_id)
+        return pd.merge(df, mol_formulas)
+
+    def load_fdr_subsample(self, instrument_settings, charge, db_id,
```
Comment (intsco) [inline app/isotope_storage.py:134]: It feels strange to see fdr terminology here as moldb is meant to operate with molecular databases and isotopic patterns.

File: app/config.py [hunk 2]
```diff
@@ -25,7 +25,12 @@
     DB_CONFIG = (POSTGRES['user'], POSTGRES['host'], POSTGRES['database'])
     DATABASE_URL = "postgresql+psycopg2://{}@{}/{}".format(*DB_CONFIG)
 
+ISOTOPE_STORAGE_DIR = CONFIG['isotope_storage']['directory']
+ISOTOPE_S3_BUCKET = CONFIG['isotope_storage'].get('s3bucket')
```
Comment (intsco) [inline app/config.py:29]: But what to do with the local VBox installation?

Comment (lomereiter) [reply app/config.py:29]: This can be set to `None`, then it will not download anything.

File: app/isotope_storage.py (hunk 1 above)
Comment (lomereiter) [reply app/isotope_storage.py:134]: Feel free to come up with a better name, but this is the best place to perform this subsampling (no need to transfer the whole table over network, wasting time on serialization)

**Pull request closed without merging**